AP_HAL_ChibiOS: add novaX AF-H7 nano flight controller (#32425)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/novaX-AF-H7_nano/README.md

# novaX AF-H7 nano Flight Controller

The novaX AF-H7 nano is a STM32H743-based flight controller by novaX.

![novaX AF-H7 nano pinout](novaX-AF-H7_nano-pinout.png)

> ## Specifications

Reviewer: get rrid of all the intternal CPU references....like addresses and buses from this

- **MCU:** STM32H743VIH6 (480MHz, 2MB Flash, TFBGA-100)
- **IMU:** Dual ICM-42688-P on SPI1 + SPI4
- **Barometer:** DPS310 or SPA06-003 on I2C2 (0x76)
- **Compass:** onboard IST8310 on I2C2 (0x0E)
- **OSD:** AT7456E (MAX7456 compatible) on SPI2
- **CAN:** FDCAN1 with TJA1051TK/3 transceiver
- **SD Card:** SDMMC1 (4-bit)
- **USB:** Type-C (OTG_FS)
- **Motor Outputs:** 10 PWM (M1-M4 bidirectional DShot capable)
- **UARTs:** 7 (USART1/2/3/6, UART4/7/8)

> - **ADC:** Dual battery voltage/current, RSSI, airspeed

Reviewer: no dual monitor, rssi or airspeed inputs

> - **LED Strip:** WS2812 on PA8

Reviewer: ```suggestion
- **LED Strip:**set up for NEOPIXEL led strings
```

## UART Mapping

> | Serial | Port   | Pins        | Default use                       |

Reviewer: ```suggestion
| Serial | Port   | Pins        | Default Protocol                      |
```
Pins column should reflect pinout image labels...not cpu pins, which have no use to a user

|--------|--------|-------------|-----------------------------------|

> | SERIAL0 | USB    | -           | MAVLink (USB)                     |

Reviewer: ```suggestion
| SERIAL0 | USB    | -           | MAVLink2 (USB)                     |
```

|---------|--------|----------------------------|------------------------|
| SERIAL0 | USB    | USB-C                      | MAVLink2               |
| SERIAL1 | USART1 | I2C/UART external pads     | MAVLink2               |

> | SERIAL2 | USART2 | DJI O3/O4 air unit         | MAVLink2               |

Reviewer: ```suggestion
| SERIAL2 | USART2 | DJI O3/O4 air unit,RX only       | MAVLink2               |
```

| SERIAL3 | USART3 | GPS                        | GPS                    |
| SERIAL4 | UART4  | OpenIPC / DJI O3 (video)   | MSP DisplayPort        |
| SERIAL6 | USART6 | ELRS / RC                  | RCIN                   |

> | SERIAL7 | UART7  | spare pads                 | MAVLink2               |

Reviewer: ```suggestion
| SERIAL7 | UART7  | spare pads                 | None       |
```

> | SERIAL8 | UART8  | spare pads                 | MAVLink2               |

Reviewer: ```suggestion
| SERIAL8 | UART8  | spare pads                 | None        |
```

## RC Input

> The RC receiver connector (silkscreen **ELRS**) is on SERIAL6 (USART6) and

Reviewer: ```suggestion
The RC receiver connector (labeled **ELRS** in pinout image) is on SERIAL6 (USART6) and
```

> defaults to `RCIN`, supporting CRSF/ELRS, SBUS and other serial protocols. To

Reviewer: ```suggestion
defaults to `RCIN` protocol, supporting CRSF/ELRS, SBUS and all other ArduPilot RC protocols, except PPM. To
```

> take RC from the DJI O3/O4 air unit on SERIAL2 instead, set `SERIAL2_PROTOCOL 23`.

Reviewer: use the SBUS input on the DJI/HD VTX interface, which is tied to USART2 RX input; you  must change the `SERIAL6_PROTOCOL` to something other than "23" and set `SERIAL2_PROTOCOL` to "23".

## GPS and Compass

The **GPS** connector is on SERIAL3 (USART3) plus I2C1 for an external compass.

> An onboard IST8310 is present on the internal I2C bus.

Reviewer: ```suggestion
An IST8310 compass is present internally, but most users will disable this compass and use an external one to avoid electrical system interference.
```

## Video / OSD

- Onboard AT7456E analog OSD (`OSD_TYPE 1`).

> - SERIAL4 (UART4) defaults to MSP DisplayPort for the OpenIPC / DJI O3 digital

Reviewer: ```suggestion
- SERIAL4 (UART4) defaults to MSP DisplayPort for DJI/OPENIPC/HD VTX connections for OSD, which can operate simultaneously with the analog OSD. SERIAL4 (UART4) is also available on a dedicated OPENIPC connector.
```

>   video link. The VTX pad exposes USART3 TX for analog VTX control if SERIAL3 is

Reviewer: ```suggestion
 The VTX connector exposes USART3 TX for analog VTX control if SERIAL3 is
```

  repurposed from GPS.

> ## Battery Monitoring

Reviewer: add a list of defaults set in hwdef for the battery monitors, I do not see any pins for BATT2 monitor....remove from hwdef

Battery monitor 1 is enabled by default with voltage and current sensing

> (divider 11.0, current scale 40.0 A/V). A second battery input (BATT2 voltage

Reviewer: ```suggestion
A second battery input (BATT2 voltage
```

and current pins) is wired but left disabled; set `BATT2_MONITOR` to enable it.

## Motor Outputs

> Outputs 1-10 are available. Outputs 1-4 support bidirectional DShot (M1-M2 on

Reviewer: ```suggestion
Outputs S1-S10 are available and support PWM and DShot. Outputs S1-S4 (also on ESC connector as M1-M4) support bidirectional DShot 
```

TIM3, M3-M4 on TIM5). M5-M6 share the TIM5 group, M7-M10 are on TIM4, and the
WS2812 LED strip is output 11 (PA8, TIM1). DShot on M7-M10 shares a DMA stream

> with the OSD SPI.

Reviewer: ```suggestion

Outputs are grouped and must all have the same output protocol within a group:

S1,2 Group 1
S3-6 Group 2
S7-10 Group3
LED Group 4, can also be used as output if SERVO11 function changed from serial led.

```

> ## Additional I/O

Reviewer: replace with a GPIO section that includes the 10 outputs and buzzer
you dont pin out the user GPIOS, remove from hwdef, nor RSSI or AS

- Analog RSSI input on ADC pin 8 (`BOARD_RSSI_ANA_PIN`); set `RSSI_TYPE 1` to use it.
- Analog airspeed input on ADC pin 4.
- Four user GPIOs, `PINIO1`-`PINIO4`, usable as relays or servo outputs.
- Buzzer output, and a GPIO to control the CAN transceiver silent pin.

> ## Building

Reviewer: remove this section, replace with a Firmware section ...see other recent hwdefs
